addons/rating/tests/test_controller.py

```diff
@@ -1,25 +1,69 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 from odoo.tests import HttpCase, tagged, new_test_user
+from odoo.addons.base.tests.common import HttpCaseWithUserDemo, HttpCaseWithUserPortal
 
 @tagged('post_install', '-at_install')
-class TestControllersRoute(HttpCase):
+class TestControllersRoute(HttpCaseWithUserDemo, HttpCaseWithUserPortal):
 
     def setUp(self):
         super(TestControllersRoute, self).setUp()
         self.user = new_test_user(self.env, "test_user_1", email="test_user_1@nowhere.com", tz="UTC")
         self.partner = self.user.partner_id
+        self.rated_partner = self.env['res.partner'].create({
+            'name': 'Test company',
+            'email': "testcompany@example.com",
+            'is_company': True,
+            'child_ids': [
+                (0, 0, {'name': 'Test child_1', 'type': 'contact', 'email': "testchild_1@example.com",}),
+            ]
+        })
 
     def test_controller_rating(self):
-
-        rating_test = self.env['rating.rating'].with_user(self.user).create({
-            'res_model_id': self.env['ir.model'].sudo().search([('model', '=', 'res.partner')], limit=1).id,
+        partner_model_id = self.env['ir.model'].sudo().search([('model', '=', 'res.partner')], limit=1).id
+        [rating_test_1, rating_test_2, rating_test_3] = self.env['rating.rating'].with_user(self.user).create([{
+            'res_model_id': partner_model_id,
```

Comment:
Use the ir.model API when possible :)
```suggestion
        [rating_test_1, rating_test_2, rating_test_3] = self.env['rating.rating'].with_user(self.user).create([{
            'res_model_id': self.env['ir.model']._get_id('res.partner'),
```